packages/core/saved-objects/core-saved-objects-server/docs/model_versions.md

```diff
@@ -1040,76 +1040,13 @@ to the `fields` option **were already present in the prior model version**. Othe
+   (*note*: both the previous and next version of Kibana must follow this rule then)
 
-(Note: this same limitation used to exist for the `update` method but has been [fixed](https://github.com/elastic/kibana/issues/165434). So while they're similar this limitation is only relevant for the `bulkUpdate` method)
+### Using `bulkUpdate` for fields with large `json` blobs
 
-The savedObjects `bulkUpdate` API is effectively a partial update (using Elasticsearch's `_update` under the hood),
-allowing API consumers to only specify the subset of fields they want to update to new values, without having to
-provide the full list of attributes (the unchanged ones). We're also not changing the `version` of the document
-during updates, even when the instance performing the operation doesn't know about the current model version
-of the document (e.g an old node during an upgrade).
-
-If this was fine before zero downtime upgrades, there is an edge case in serverless when this API is used
-to update fields that are the "source" of another field's backfill that can potentially lead to data becoming inconsistent.
-
-For example, imagine that:
-
-1. In model version 1, we have some `index (number)` field.
-
-2. In model version 2, we introduce a `odd (boolean)` field that is backfilled with the following function:
-
-```ts
-let change: SavedObjectsModelDataBackfillChange = {
-  type: 'data_backfill',
-  backfillFn: (doc, ctx) => {
-    return { attributes: { odd: doc.attributes.index % 2 === 1 } };
-  },
-};
-```
-
-3. During the cohabitation period (upgrade), an instance of the new version of Kibana creates a document 
-
-E.g with the following attributes:
-
-```ts
-const newDocAttributes = {
-  index: 12,
-  odd: false,
-}
-```
-
-4. Then an instance of the old version of Kibana updates the `index` field of this document
-
-Which could occur either while being still in the cohabitation period, or in case of rollback:
-
-```ts
-savedObjectClient.bulkUpdate({
-  objects: [{
-    type: 'type', 
-    id: 'id', 
-    attributes: {
-      index: 11
-    }
-  }]
-});
-```
-
-We will then be in a situation where our data is **inconsistent**, as the value of the `odd` field wasn't recomputed:
-
-```json
-{
-  index: 11,
-  odd: false,
-}
-```
-
-The long term solution for that is implementing [backward-compatible updates](https://github.com/elastic/kibana/issues/165434), however
-this won't be done for the MVP, so the workaround for now is to avoid situations where this edge case can occur.
-
-It can be avoided by either:
-
-1. Not having backfill functions depending on the value of the existing fields (*recommended*)
-
-2. Not performing update operations impacting fields that are used as "source" for backfill functions
-   (*note*: both the previous and next version of Kibana must follow this rule then)
\ No newline at end of file
+The savedObjects `bulkUpdate` API will update documents client-side and then reindex the updated documents.
```

Comment:
These are really useful docs, but I think developers are more likely to read it if it was part of the API documentation on the `bulkUpdate` method?

Reply:
good call, I'll add it there too.